public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -198,6 +199,15 @@ export function initDashboard(args: InitDashboardArgs): ThunkResult<void> {
       console.log(err);
     }
 
+    const left = storeState.explore && storeState.explore.left;
+    if (left && left.originPanelId) {
+      const panelArrId = dashboard.panels.findIndex(panel => panel.id === left.originPanelId);
+      if (panelArrId > -1) {
+        dashboard.panels[panelArrId].targets = left.queries.map(query => ({ refId: query.refId, expr: query.expr }));
+      }
+      dashboard.startRefresh();
+      dispatch(resetExploreAction({ force: true }));
```

Comment:
Why do we need to reset Explore here?

Reply:
So, for now the way I've implemented this feature is that when returning to the origin panel from explore, if we're doing so with changes all the explore state is reset (as is the behavior in master currently) _except_ the queries and the origin panel ID. Then after the dashboard has been initialized with the queries from explore, we completely reset the explore state (with `{ force: true }`), so that on subsequent dashboard loads we aren't taking the modified queries from explore again.